Formula/ghc.rb

```diff
@@ -86,14 +105,27 @@ def install
 
       ENV.prepend_path "PATH", binary/"bin"
     end
+    resource("cabal-install").stage { (binary/"bin").install "cabal" }
+    system "cabal", "v2-update"
```

Comment:
Could alternatively use `hadrian` bootstrap (https://downloads.haskell.org/~ghc/9.4.2/hadrian-bootstrap-sources/) which doesn't need `cabal` to get dependencies.

Adding `cabal` does potentially allow us to build HEAD (testing it out locally) since it makes it easier to obtain `alex` and `happy`.